docs/proposals/20-ha-control-planes.md

```diff
@@ -528,10 +529,11 @@ In this option existing recovery mechanisms as described above are used. There i
 _Pros:_
 * Less complex to implement since no dynamic re-balancing of pods is required and there is no need to determine if there is an AZ outage.
 * Additional cost to host an HA shoot control plane is kept to the bare minimum.
-* Existing recovery mechanisms are leveraged.
+* Existing recovery mechanisms are leveraged:
+  * `kube-controller-manager` will create new replicas for the affected Deployment Pods and they will recover in short period of time.
+  * The replacement Pods for the affected StatefulSet Pods will fail to be scheduled due to their affinity rules (etcd Pods) or volume requirements (prometheus and loki Pods) to run in the outage zone and will not recover.
 
 _Cons:_
-* Existing recovery of pods will result in a downtime of up to a total of [--machine-health-timeout](https://github.com/gardener/gardener-extension-provider-gcp/blob/dff3d2417ff732fcce69ba10bbe5d04e13781539/charts/internal/machine-controller-manager/seed/templates/deployment.yaml#L49) + [--machine-drain-timeout](https://github.com/gardener/gardener-extension-provider-gcp/blob/dff3d2417ff732fcce69ba10bbe5d04e13781539/charts/internal/machine-controller-manager/seed/templates/deployment.yaml#L48) (which as of today is 2hr10min) for pods which can be rescheduled onto another node/zone. Stateful pods due to the affinity rules will not recover, however stateless pods will eventually recover.
```

Comment:
@unmarshall I decided to remove this point as it does not make sense to me. Unhealthy Nodes should be drained, instead they should be deleted after `--machine-health-timeout` (10min by default). Hence the recovery time should not be > 2h. Let me know if I am missing something.

Reply:
the reasons of unhealthiness like `nodePressure` and `diskPressure` would mean that the pods would still be running on the node (except maybe a few) , so we should be doing graceful draining of node in such scenario respecting their PDBs for service availability. I am looking at it more from the user point of view when they run their workload on a cluster.
only in scenarios like `readOnlyfileSystem`(pod can't interact with fs) or `kubelet gone for too long` (eviction can't go through) are the cases where we skip drain to avoid wasting time on drain

Reply:
Thanks for explaining. In case of a zone outage I think that the Nodes will be `NotReady`/`Unknown` for a reason that is not NodePressure/DiskPressure. In the described simulations we forbid ingress/egress traffic for a zone and kubelets cannot report that they are healthy. So in this case the drain will be skipped.
Anyways, I reworked this point and removed the part with the `--machine-drain-timeout`. In today's head, it is written like it will always result in 2hr10min which should not be true.
tl;dr: If the Node is `NotReady` (except conditions like NodePressure/DiskPressure) the drain is skipped. In case of NodePressure/DiskPressure I would expect then the drain to be able to complete and not to get stuck.